### PR TITLE
multiple embed blocks fix

### PR DESCRIPTION
**Pull request by vcraescu**
Embed multiple blocks it's not working when you use the default prefix and postfix: %embed-block|, |end%. 
Also, preg_* functions should be avoided parsing large files due their length limit.

### CODE REVIEW

File: Templating/Helper/CmfBlockHelper.php [hunk 1]
```diff
@@ -92,17 +91,20 @@ public function getName()
     /**
      * Executes the block as specified in the content.
      *
-     * @param array $block An array including the block name
+     * @param $name
      *
-     * @return string the rendered block
+     * @return null|string|\Symfony\Component\HttpFoundation\Response
```
Comment (dbu) [inline Templating/Helper/CmfBlockHelper.php]: please use `use` statements for annotations as well. is this the return type of the render method?

File: Templating/Helper/EmbedBlocksParser.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        return $this->prefix;
+    }
+
+    /**
+     * @param mixed $prefix
+     *
+     * @return EmbedBlocksParser
+     */
+    public function setPrefix($prefix)
```
Comment (dbu) [inline Templating/Helper/EmbedBlocksParser.php]: why would you need to change this? i think having it in the constructor is enough.

File: Templating/Helper/EmbedBlocksParser.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    public function __construct($prefix, $postfix)
+    {
+        $this->prefix = $prefix;
+        $this->postfix = $postfix;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getPrefix()
```
Comment (dbu) [inline Templating/Helper/EmbedBlocksParser.php]: does anything outside the parser care about this value?

File: Templating/Helper/CmfBlockHelper.php [hunk 4]
```diff
@@ -27,20 +27,17 @@ class CmfBlockHelper extends Helper
...
-    private $postfix;
+    private $parser;
 
     /**
      * @var LoggerInterface
      */
     private $logger;
 
-    public function __construct(SonataBlockHelper $sonataBlock, $prefix, $postfix, LoggerInterface $logger = null)
+    public function __construct(SonataBlockHelper $sonataBlock, EmbedBlocksParser $parser, LoggerInterface $logger = null)
```
Comment (dbu) [inline Templating/Helper/CmfBlockHelper.php]: this is a BC break. luckily, we are in the process of going to version 2, so its ok to break backwards compatibility.

but please add a note to CHANGELOG.md that mentions this change.

**Comment by dbu** [comment]
awesome, great work! i just have a few nitpiks that i commented on, otherwise looks good.

**Comment by craescu** [comment]
Hi David!

I'll do another PR during the weekend with this fix which will not break 1.x interface.

Thanks!

**Comment by dbu** [comment]
for 2.0 we perfer to do the BC break to clean things up, so this code looks fine. the BC solution for 1.x can be to simply instantiate the parser within the helper instead of using DI, right?

though, for the sake of the git history and keeping master in sync with the 1.x branch, i think its best if you cherry-pick this commit into a branch off the 1 branch, do that PR and once we merge this, i update master from 1. then you can rebase this PR and it will not add the whole parser but just fix the setup and service. does that make sense?

**Comment by craescu** [comment]
I was thinking to move the the entire parse code directly into the helper for 1.x version. Version 2 will be just a refact of this code where parse gets its own class. If we instantiate the parser with the helper then in version 2 we will have 2 compatibility breaks. Do you agree?

**Comment by dbu** [comment]
sounds like a good plan to me. so i wait with merging this until you
ported it to the 1.3 branch (we don't maintain the older branches anymore)

**Comment by craescu** [comment]
Sure. I'll do it this weekend.

**Comment by dbu** [comment]
next step: #255 and then rebase this branch on master.

**Comment by dbu** [comment]
#253 is now in master. please rebase this PR and fix the conflicts, then we can merge this.

**Comment by craescu** [comment]
I will do it this weekend. Sorry I had a crazy week and didn't get a chance to take care of this.

**Comment by vcraescu** [comment]
Can you help me out with these build errors?

File: Resources/config/services.xml [hunk 5]
```diff
@@ -4,67 +4,85 @@
...
+        <parameter key="cmf_block.service.reference.class">Symfony\Cmf\Bundle\BlockBundle\Block\ReferenceBlockService</parameter>
+        <parameter key="cmf_block.service.action.class">Symfony\Cmf\Bundle\BlockBundle\Block\ActionBlockService</parameter>
+        <!--container block service is reused for slideshows -->
+        <parameter key="cmf_block.service.slideshow.class">Symfony\Cmf\Bundle\BlockBundle\Block\ContainerBlockService</parameter>
+    </parameters>
+
     <services>
 
-        <service id="cmf.block.simple" class="Symfony\Cmf\Bundle\BlockBundle\Block\SimpleBlockService">
+        <service id="cmf.block.simple" class="%cmf_block.service.simple.class%">
```
Comment (dbu) [inline Resources/config/services.xml]: something went wrong here. we removed the parameters for classes, as they are deprecated by symfony itself.

**Comment by dbu** [comment]
hm, that error seems unrelated to your changes. we also see it in master. probably changes in one of our also-not-released dependencies that prevent installing the bundle. i will have a look.

File: Resources/config/services.xml (hunk 5 above)
Comment (vcraescu) [inline Resources/config/services.xml]: Sorry missed that. I'll fix it.

**Comment by dbu** [comment]
okay, fixed the build in #256. can you please rebase this branch on master? it should then be green again.

**Comment by vcraescu** [comment]
Why did you merge this into master? Any issues?

**Comment by dbu** [comment]
thanks! sorry, did not see the notification for your push.

**Comment by vcraescu** [comment]
Np. Thanks! :)

**Comment by dbu** [comment]
thank you for the contribution!

**Comment by vcraescu** [comment]
uw!

**Pull request merged**